File: auto/feature_def/auto_common_spec/extensions/genPropBag/ext/platform/android/src/com/rho/genpropbag/GenPropBag.java
package com.rho.genpropbag;

import java.util.Map;

import com.rhomobile.rhodes.api.IMethodResult;
import com.rhomobile.rhodes.api.MethodResult;

import com.rhomobile.rhodes.Logger;

public class GenPropBag extends GenPropBagBase implements IGenPropBag {

	private MethodResult callback = null;
    
    public GenPropBag(String id) {
        super(id);
    }

    public void setCallbackProp(IMethodResult result) {
    	if (result.hasCallback())
    	{
    		Logger.I("GenPropBag", "setCallback");
			((MethodResult) result).keepAlive();
			callback = ((MethodResult) result);
    	}
    }
    public void callCallback(IMethodResult result) {
    	if (callback != null && callback.hasCallback()) {
    		Logger.I("GenPropBag", "FireCallback");
    		callback.set("set");
    	}
    }

}

File: auto/feature_def/auto_common_spec/extensions/genPropBag/ext/platform/iphone/impl/GenPropBag.m

#import "GenPropBag.h"


@implementation GenPropBag

-(id)init {
    self = [super init];
    callbackHolder = nil;
    return self;
}

-(void) setCallbackProp:(id<IMethodResult>)methodResult {
    callbackHolder = methodResult;
}

-(void) callCallback:(id<IMethodResult>)methodResult {
    if ([callbackHolder hasCallback]) {
        [callbackHolder setResult:@"set"];
    }
}


@end